crates/lang_srv/src/server.rs

```diff
@@ -1,34 +1,41 @@
 use analysis::HIGHLIGHT_TOKENS_LEGEND;
-use parking_lot::{Mutex, MutexGuard};
-use registry::{DocumentChange, Registry};
+use log::{debug, trace};
+use registry::Registry;
+use std::future::Future;
+
+use std::time::Duration;
 use tower_lsp::jsonrpc::Result;
 use tower_lsp::lsp_types::*;
 use tower_lsp::{Client, LanguageServer, LspService, Server};
 
+use crate::analysis::global_analysis;
+
 mod analysis;
 mod convert;
 mod registry;
 
 #[derive(Debug)]
 struct RocLs {
+    pub inner: Inner,
     client: Client,
-    registry: Mutex<Registry>,
+}
+
+///This exists so we can test most of RocLs without anything LSP related
+#[derive(Debug)]
+struct Inner {
+    registry: Registry,
 }
```

Comment:
Nit: Let's call this Manager and the outer one Server or something. Or at least something better than RocLs/Inner.